langchain4j-core/src/main/java/dev/langchain4j/rag/content/retriever/WebSearchContentRetriever.java

```diff
@@ -0,0 +1,139 @@
+package dev.langchain4j.rag.content.retriever;
+
+import dev.langchain4j.data.document.DocumentSplitter;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.inmemory.InMemoryEmbeddingStore;
+import dev.langchain4j.web.search.WebSearchResults;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.rag.content.Content;
+import dev.langchain4j.rag.query.Query;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.web.search.WebSearchEngine;
+import lombok.Builder;
+
+import java.util.List;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.*;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * A {@link ContentRetriever} backed by a {@link WebSearchEngine}.
+ * <br>
+ * By default, this retriever fetches the  all web pages {@link Content} for a given {@link Query}.
+ * <br>
+ * <br>
+ * Configurable parameters (optional):
+ * <ul>
+ *     <li>{@link DocumentSplitter} - To split the web pages into smaller text segments.</li>
+ *     <li>{@link EmbeddingModel} - To generate embeddings to ingest them into an internal {@link EmbeddingStore} and find the most
+ *     relevant web search results based on the {@link Query}</li>
+ *     <li>{@link #maxResults} - The maximum number of results to retrieve by the internal {@link EmbeddingStore}.
+ *     <br>
+ *     Default value is 3.
+ *     </li>
+ * </ul>
+ */
+public class WebSearchContentRetriever implements ContentRetriever{
+
+    private static final int DEFAULT_MAX_RESULTS = 3;
+    
+    private final WebSearchEngine webSearchEngine;
+    private final EmbeddingModel embeddingModel;
+    private final DocumentSplitter documentSplitter;
+    private final EmbeddingStore<TextSegment> embeddingStore;
```

Comment:
We should create a new `InMemoryEmbedding` store for each `retrieve()` call, otherwise later searches will return some content from previous searches. So it should be a local field in the method, not a class field.